ui/lib/services/bom_bar_client.dart

```diff
@@ -38,9 +38,10 @@ class BomBarClient {
     ));
   }
 
-  /// Returns all available projects.
-  Future<List<Project>> getProjects() async {
-    final response = await dio.getUri(_projectsUrl);
+  /// Returns all or projects matching the provided [filter] fragment.
+  Future<List<Project>> findProjectsBySearchFragment(String? filter) async {
```

Comment:
Dart has a syntax to pass an optional parameter, which seems more fitting here: `foo([String? filter])`.

(You can then call this function without parameter to indicate there is no fragment.)